services/swagger.js

```diff
@@ -151,20 +151,20 @@ function flattenModelDefinitions(definitions) {
     });
 }
 
-function flattenModel(item, props) {
+function mergeToFlatModel(flatModel, model) {
     //have to convert 'required' array to object for merging purposes
-    if (item.required && Array.isArray(item.required)) {
-        var requiredObj = {};
-        item.required.reduce(function (prevVal, currentVal, idx) {
-            prevVal[currentVal] = idx;
-            return prevVal;
-        }, requiredObj);
-        item.required = requiredObj;
+    if (model.required && Array.isArray(model.required)) {
+        var requiredAsObj = {};
```

Comment:
right, but it wouldn't be until it completes ...
right?
can you try [the code I wrote](https://github.com/BlueOakJS/blueoak-server/pull/67#discussion_r79875209)? if that doesn't work, I want to know why so I stop writing bad code :-)